woodwork/tests/logical_types/test_logical_types.py

```diff
@@ -659,3 +664,42 @@ def test_null_invalid_postal_code_numeric():
     expected = pd.DataFrame({"data": data})
     df.ww.init(logical_types=types, null_invalid_values=True)
     pd.testing.assert_frame_equal(df, expected)
+
+
+@pytest.mark.parametrize("null_type", [None, pd.NaT, np.nan, "null", "N/A", "mix", 5])
+@pytest.mark.parametrize("data_type", [int, float])
+def test_integer_nullable(data_type, null_type):
+    nullable_nums = pd.DataFrame(
+        np.random.choice([data_type(i) for i in range(10)], 100), columns=["num_nulls"]
+    )
+    nullable_nums["num_nulls"].iloc[-5:] = (
+        [None, pd.NA, np.nan, "NA", "none"] if null_type == "mix" else [null_type] * 5
+    )
+    nullable_nums.ww.init()
+
+    if null_type != 5:
+        assert isinstance(nullable_nums.ww.logical_types["num_nulls"], IntegerNullable)
+        assert all(nullable_nums["num_nulls"][-5:].isna())
+    elif data_type is int:
+        assert isinstance(nullable_nums.ww.logical_types["num_nulls"], Integer)
+    else:
+        assert isinstance(nullable_nums.ww.logical_types["num_nulls"], Double)
+
```

Comment:
What happens when we mix integers and booleans together?  What would we expect to happen?

Reply:
If it's just integers and booleans, we would expect the same behaviour which is an inference of `Categorical` based on the `dtype: object`. If `nan` values are added it would be the same.

Reply:
Same thing with only 1's and 0's? That's a case we'll want to be inferred as Boolean at some point.